sudachi/src/analysis/morpheme_list.rs

```diff
@@ -0,0 +1,213 @@
+/*
+ *  Copyright (c) 2021 Works Applications Co., Ltd.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+use std::ops::Deref;
+
+use crate::analysis::node::Node;
+use crate::dic::grammar::Grammar;
+// use crate::analysis::morpheme::Morpheme;
+use crate::analysis::stateless_tokenizer::DictionaryAccess;
+use crate::dic::lexicon::word_infos::WordInfo;
+use crate::prelude::*;
+
+pub struct MorphemeList<'a, T> {
+    pub dict: T,
+    pub input_text: &'a str,
+    pub path: Vec<Node>,
+}
+
+impl<'a, T> MorphemeList<'a, T>
+where
+    T: Deref,
+    <T as Deref>::Target: DictionaryAccess,
+{
+    pub fn new(dict: T, input_text: &'a str, path: Vec<Node>) -> Self {
+        Self {
+            dict,
+            input_text,
+            path,
+        }
+    }
+
+    pub fn iter(&self) -> MorphemeIter<T> {
+        MorphemeIter {
+            list: &self,
+            index: 0,
+        }
+    }
+}
+
+impl<T> MorphemeList<'_, T>
+where
+    T: Deref,
+    <T as Deref>::Target: DictionaryAccess,
+{
+    fn get_node(&self, index: usize) -> &Node {
+        &self.path[index]
+    }
+
+    fn get_grammar(&self) -> &Grammar {
+        self.dict.grammar()
+    }
+
+    fn get_begin(&self, index: usize) -> usize {
+        self.path[index].begin
+    }
+
+    fn get_end(&self, index: usize) -> usize {
+        self.path[index].end
+    }
+
+    fn get_surface(&self, index: usize) -> &String {
```

Comment:
this should be `&str`. Python wrapper should explicitly convert `&str` to PyString